lib/jsonapi-serializers/serializer.rb

```diff
@@ -48,19 +52,22 @@ def id
       # per the spec naming recommendations: http://jsonapi.org/recommendations/#naming
       # For example, 'MyApp::LongCommment' will become the 'long-comments' type.
       def type
-        object.class.name.demodulize.tableize.dasherize
+        class_name = object.class.name
+        @@class_names[class_name] ||= class_name.demodulize.tableize.dasherize
```

Comment:
I think you could safely add a `.freeze` to the end of all of these values, here and below, which would "guarantee" the safety that this caching system will never store strings that might mutate. I can't think of a case in which they would be mutated, but to be safe.